submissions/evgeniy241984/a-tiny-JS-world/index.js

```diff
@@ -0,0 +1,102 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+"use strict";
+
+class Animals {
+  constructor(gender, name, saying) {
+    this.gender = gender;
+    this.name = name;
+    this.saying = saying;
+  }
+
+  say() {
+    return `i'm ${this.species}; My name is: <strong> ${this.name} </strong>; My gender: ${this.gender}; Saying: <em> ${this.saying} </em>`;
+  }
+}
+
+class Carnivora extends Animals {
+  constructor(gender, name, saying, paws) {
+    super(gender, name, saying, paws);
+    this.paws = paws || 4;
+  }
+  say() {
+    return super.say() + `; I have ${this.paws} paws`;
+  }
+}
+
+class Dog extends Carnivora {
+  constructor(name, saying, gender, paws) {
+    super(gender, name, saying, paws);
+    this.species = "dog";
+  }
+}
+
+class Cat extends Carnivora {
+  constructor(name, saying, gender, paws) {
+    super(gender, name, saying, paws);
+    this.species = "cat";
+  }
+}
+
+class Homo extends Animals {
+  constructor(gender, name, sayinng, legs, hands) {
+    super(gender, name, sayinng, legs, hands);
+    this.hands = hands || 2;
+    this.legs = legs || 2;
+    this.species = "human";
+  }
+  say() {
+    return super.say() + `; I have ${this.hands} legs, ${this.legs} hands`;
+  }
+}
+
+class Man extends Homo {
+  constructor(name, saying, gender, legs, hands) {
+    super(gender, name, saying, legs, hands);
+    this.gender = "male";
+  }
+}
+
+class Woman extends Homo {
+  constructor(name, saying, gender, legs, hands) {
+    super(gender, name, saying, legs, hands);
+    this.gender = "female";
```

Comment:
Constructor receives `undefined` as `gender` and so it is passed further down via `super` call. Then `this.gender` gets explicitly assigned.
1. What expecting `gender` via constructor call while it is ignored?
2. Why not to pass gender explicitly to `super`?